api/client_agent.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/mattermost/mattermost-load-test-ng/loadtest"
+)
+
+// Agent represents a load-test agent.
+// It exposes methods to manage a load-test agent resource through API.
+type Agent struct {
+	id     string
+	apiURL string
+	client *Client
+}
+
+// agentResponse contains the data returned by the load-test agent API.
+type agentResponse struct {
+	Id      string           `json:"id,omitempty"`      // The load-test agent unique identifier.
+	Message string           `json:"message,omitempty"` // Message contains information about the response.
+	Status  *loadtest.Status `json:"status,omitempty"`  // Status contains the current status of the load test.
+	Error   string           `json:"error,omitempty"`   // Error is set if there was an error during the operation.
+}
+
+func (a *Agent) apiRequest(req *http.Request) (agentResponse, error) {
```

Comment:
Why don't you used `*agentResponse` on returns?

Reply:
Because I didn't need that to be a pointer. When possible, and if it makes sense, I prefer to use value semantics. It's less prone to due to races and side effects.

Reply:
Sounds fair 👍